packages/data/src/graphql/tokenListingCounts/tokenListingCountsByMetaId.ts

```diff
@@ -0,0 +1,55 @@
+
+import { NftTokensAggregate } from '../codegen/graphql';
+import { fetchGraphQl } from '../fetch';
+import { tokenListingCountsByMetaIdQuery } from './tokenListingCountsByMetaId.query';
+
+export type TokenListingCounts = {
+  totalTokensCount: number;
+  simpleListingCount: number;
+  auctionListingsCount: number;
+  totalListingsCount: number;
+  displayTotalListings: number;
+}
+
+export type TokenListingQueryResults = {
+  tokensCount: NftTokensAggregate;
+  simpleListingsCount: NftTokensAggregate;
+  auctionListingsCount: NftTokensAggregate;
+}
+
+export const tokenListingCountsByMetaId = async (
+  metadataId: string,
+): Promise<TokenListingCounts> => {
+  const { data, error } = await fetchGraphQl<TokenListingQueryResults>({
+    query: tokenListingCountsByMetaIdQuery,
+    variables: {
+      metadataId,
+    },
+  });
+
+  // log and rethrow for now...
+  // question: do we want to use the same { error, data } pattern in these
+  // methods as well?
+  if (error) {
```

Comment:
See the description for conversation around this.  Almost wondering if we offer two methods, one which will throw and one that will offer the error, data result shape 🤔